main.go

```diff
@@ -6,34 +6,84 @@ import (
 
 	"github.com/asecurityteam/awsconfig-filterd/pkg/filter"
 	v1 "github.com/asecurityteam/awsconfig-filterd/pkg/handlers/v1"
+	"github.com/asecurityteam/components"
 	"github.com/asecurityteam/runhttp"
 	"github.com/asecurityteam/serverfull"
 	"github.com/asecurityteam/settings"
 )
 
-func main() {
-	source, err := settings.NewEnvSource(os.Environ())
+type Config struct {
+	Filter     *filter.FilterConfig
+	Producer   *components.ProducerConfig
+	LambdaMode bool `description:"Use the Lambda SDK to start the system."`
+}
+
+func (*Config) Name() string {
+	return "awsconfigfilterd"
+}
+
+type Component struct {
+	Filter   *filter.FilterComponent
+	Producer *components.ProducerComponent
+}
+
+func NewComponent() *Component {
+	return &Component{
+		Filter:   filter.NewFilterComponent(),
+		Producer: components.NewProducerComponent(),
+	}
+}
+
+func (c *Component) Settings() *Config {
+	return &Config{
+		Filter:   c.Filter.Settings(),
+		Producer: c.Producer.Settings(),
+	}
+}
+
+func (c *Component) New(ctx context.Context, conf *Config) (func(context.Context, settings.Source) error, error) {
+	f, err := c.Filter.New(ctx, conf.Filter)
 	if err != nil {
-		panic(err.Error())
+		return nil, err
 	}
-	ctx := context.Background()
-	resourceFiltererComponent := &filter.ResourceTypeFiltererComponent{}
-	resourceTypeFilterer := new(filter.ResourceTypeFilterer)
-	err = settings.NewComponent(ctx, source, resourceFiltererComponent, resourceTypeFilterer)
+	p, err := c.Producer.New(ctx, conf.Producer)
 	if err != nil {
-		panic(err.Error())
+		return nil, err
 	}
-	configFilterHandler := v1.ConfigFilterHandler{
+
+	filterHandler := &v1.ConfigFilter{
 		LogFn:          runhttp.LoggerFromContext,
 		StatFn:         runhttp.StatFromContext,
-		ConfigFilterer: resourceTypeFilterer,
+		ConfigFilterer: f,
+		Producer:       p,
 	}
 	handlers := map[string]serverfull.Function{
-		"filter": serverfull.NewFunction(configFilterHandler.Handle),
+		"filter": serverfull.NewFunction(filterHandler.Handle),
 	}
-
 	fetcher := &serverfull.StaticFetcher{Functions: handlers}
-	if err := serverfull.Start(ctx, source, fetcher); err != nil {
+	if conf.LambdaMode {
+		return func(ctx context.Context, source settings.Source) error {
+			return serverfull.StartLambda(ctx, source, fetcher, "filter")
+		}, nil
+	}
+	return func(ctx context.Context, source settings.Source) error {
+		return serverfull.StartHTTP(ctx, source, fetcher)
+	}, nil
+}
+
+func main() {
+	source, err := settings.NewEnvSource(os.Environ())
+	if err != nil {
+		panic(err.Error())
+	}
+	ctx := context.Background()
+	runner := new(func(context.Context, settings.Source) error)
```

Comment:
Question: How does this work/what is it doing? It looks like you're creating an empty function signature here.

Reply:
The `new` built-in creates a pointer to an new memory allocation that contains the zero value of a type. The function signature is an instance of a type. For example, we can do things like this:

```golang
func takesAFunc(fn func() error) (func() error, error) {
  return fn
}

type AFunction func() error

var fn func() error
```

When you omit the body of the function is becomes a type like any other in the language. In this case, we're creating a new pointer to a function type with a specific signature. Since the component we're loading returns a function with this signature then we can pass the new pointer into the loader. At the end of loading, the pointer now points to an actual instance of the function signature rather than the zero value.

Reply:
This makes sense to me. It might be helpful to add a comment here, as the magic that makes this work is inside the settings library.